workflows/metatdenovo.nf

```diff
@@ -73,17 +144,373 @@ workflow METATDENOVO {
+        .set { ch_featurecounts }
+
+    ch_collect_stats
+        .combine(BAM_SORT_STATS_SAMTOOLS.out.idxstats.collect { meta, idxstats -> idxstats }.map { [ it ] } )
+        .set { ch_collect_stats }
+
+    FEATURECOUNTS_CDS ( ch_featurecounts)
+    ch_versions       = ch_versions.mix(FEATURECOUNTS_CDS.out.versions)
+
+    //
+    // MODULE: Collect featurecounts output counts in one table
+    //
+    FEATURECOUNTS_CDS.out.counts
+        .collect() { meta, featurecounts -> featurecounts }
+        .map { featurecounts -> [ [ id:"${assembler}.${orf_caller}" ], featurecounts ] }
+        .set { ch_collect_feature }
+
+    COLLECT_FEATURECOUNTS ( ch_collect_feature )
+    ch_versions           = ch_versions.mix(COLLECT_FEATURECOUNTS.out.versions)
+    ch_fcs_for_stats      = COLLECT_FEATURECOUNTS.out.counts.collect { meta, tsv -> tsv }.map { [ it ] }
+    ch_fcs_for_summary    = COLLECT_FEATURECOUNTS.out.counts.map { meta, tsv -> tsv }
+    ch_collect_stats
+        .combine(ch_fcs_for_stats)
+        .set { ch_collect_stats }
+
+    //
+    // SUBWORKFLOW: run eggnog_mapper on the ORF-called amino acid sequences
+    //
+    if ( ! params.skip_eggnog ) {
+        EGGNOG(params.eggnog_dbpath, ch_protein, ch_fcs_for_summary )
+        ch_versions = ch_versions.mix(EGGNOG.out.versions)
+        ch_merge_tables = EGGNOG.out.sumtable
+    } else {
+        ch_protein
+            .map { meta, protein -> [ meta, [] ] }
+            .set { ch_merge_tables }
+    }
+
+
+    //
+    // SUBWORKFLOW: run kofamscan on the ORF-called amino acid sequences
+    //
+    if( !params.skip_kofamscan ) {
+        ch_protein
+            .map { meta, protein -> [ meta, protein ] }
+            .set { ch_kofamscan }
+        KOFAMSCAN( ch_kofamscan, ch_fcs_for_summary)
+        ch_versions = ch_versions.mix(KOFAMSCAN.out.versions)
+        ch_kofamscan_summary = KOFAMSCAN.out.kofamscan_summary.collect().map { meta, tsv -> tsv }
+        ch_merge_tables
+            .combine( ch_kofamscan_summary )
+            .set { ch_merge_tables }
+    } else {
+        ch_merge_tables
+            .map { meta, tsv -> [ meta, tsv, [] ] }
+            .set { ch_merge_tables }
+    }
+
+    // set up contig channel to use in CAT and TransRate
+    UNPIGZ_CONTIGS(ch_assembly_contigs)
+    ch_unzipped_contigs = UNPIGZ_CONTIGS.out.unzipped
+    ch_versions = ch_versions.mix(UNPIGZ_CONTIGS.out.versions)
+
+    //
+    // MODULE: Use TransRate to judge assembly quality, piped into MultiQC
+    //
+    TRANSRATE(ch_unzipped_contigs)
+    ch_versions = ch_versions.mix(TRANSRATE.out.versions)
+
+    //
+    // SUBWORKFLOW: Eukulele
+    //
+    ch_eukulele_db = Channel.empty()
+    if( !params.skip_eukulele){
+        // Create a channel for EUKulele either with a named database or not. The latter means a user-provided database in a directory.
+        if ( params.eukulele_db ) {
+            Channel
+                .of ( params.eukulele_db.split(',') )
```

Comment:
I want to point out you're taking advantage of a Nextflow quirk (since it's not a confirmed bug). 
`split` returns a `String[]`, where as `tokenize` returns a `List`. 
`Channel.of` in this case should be returning an array here, but it's separating the elements because it's receives an `Object[]` rather than a `List`.
The factory you should use in this case is `Channel.fromList`.